crates/wasi-common/src/sys/windows/hostcalls_impl/fs.rs

```diff
@@ -79,8 +79,23 @@ pub(crate) fn fd_fdstat_get(fd: &File) -> Result<wasi::__wasi_fdflags_t> {
     Ok(fdflags)
 }
 
-pub(crate) fn fd_fdstat_set_flags(fd: &File, fdflags: wasi::__wasi_fdflags_t) -> Result<()> {
-    unimplemented!("fd_fdstat_set_flags")
+pub(crate) fn fd_fdstat_set_flags(fd: &mut File, fdflags: wasi::__wasi_fdflags_t) -> Result<()> {
+    let handle = unsafe { fd.as_raw_handle() };
+
+    let access_mode = winx::file::query_access_information(handle)?;
+
+    let access_mode = file_access_mode_from_fdflags(
+        fdflags,
+        (access_mode & AccessMode::FILE_READ_DATA).bits() != 0,
```

Comment:
You can use [`contains()`](https://docs.rs/bitflags/1.2.1/bitflags/#methods-1)

Reply:
I'll fix.